### ISSUE TITLE
Methods where all paths allocate should be shown as allocations (optional)

### ISSUE THREAD

**Issue by viktor-svub** [controlflow/resharper-heapview#16]
If this is possible with R# engine... Currently a simple generic static factory hides the allocation.

**Comment by ltrzesniewski** [comment]
Maybe it would be easier to support a manual annotation, like an `[Allocates]` attribute on a method which allocates (using the same annotation mechanism than other R# attributes like `[NotNull]`).

**Comment by viktor-svub** [comment]
I would expect the manual annotation way to be very brittle, depending on the author to correctly inspect all paths in the fist place, and 3rd-party annotations being invalidated by allocations optimized away, e.g. when rewriting a yield enumerator to a struct-based one.

**Comment by ltrzesniewski** [comment]
Yes, I wouldn't put these annotations everywhere. They'd just be for a few methods I know to be expensive. And for those I'd prefer a false positive than a false negative anyway.
